dashboard/app/controllers/courses_controller.rb

```diff
@@ -53,6 +55,12 @@ def show
       return
     end
 
+    # If this is a single-unit course, redirect to the unit overview
+    if @unit_group.single_unit_course?
+      redirect_to "/s/#{@unit_group.default_units.first.name}/"
```

Comment:
please use the rails path helpers when possible:
```suggestion
      redirect_to script_path(@unit_group.default_units.first)
```
that will make @daynew 's job easier soon.